src/schema/index.ts

```diff
@@ -39,4 +40,7 @@ export type StepDependsOn = Extract<
 > extends (infer U)[]
   ? U
   : never;
+
 export type Field = Fields extends (infer U)[] ? U : never;
+
+export type PipelineNotify = BuildNotify extends (infer U)[] ? U : never;
```

Comment:
🤔 I'm not super keen on exposing all these, not sure what to do tho, maybe I should create an internal types file??

Reply:
What's the disadvantage in exporting these? Are they likely to be used?

Reply:
Probably not very frequently used but I'm more worried that they might change frequently based off the JSON schema and the naming of these isn't particularly consistent.